docs/advanced-data-preprocessing.md

```diff
@@ -0,0 +1,74 @@
+# Advanced Data Processing
+Our library also supports a powerful data processing backed which can be used by the users to perform custom data preprocessing including
+1. Support for multiple datasets
+1. Creating custom data processing pipeline for the datasets.
+1. Combining multiple datasets into one, even if they have different formats.
+1. Mixing datasets as requried and sampling if needed each with different weights.
+
+These things are supported via what we call a [`data_config`](#data-config) which can be passed an an argument to sft trainer.
+
+## Data Config
+
+Data config is a YAML configuration file which users can provide to `sft_trainer.py`, in this file they can include a variety of datasets and configurations. This data config is passed to `sft_trainer` through the `--data_config` flag.
```

Comment:
Optional suggestion: 

**Data config is a YAML configuration file which users can provide to `sft_trainer.py`. This file allows users to include multiple datasets and configurations and is passed to `sft_trainer.py` using the `--data_config` flag.**